src/test/java/teammates/test/cases/BaseTestCaseWithDatastoreAccess.java

```diff
@@ -187,6 +187,7 @@ private void equalizeIrrelevantData(AccountAttributes expected, AccountAttribute
                 expected.studentProfile.googleId = actual.googleId;
             }
             expected.studentProfile.modifiedDate = actual.studentProfile.modifiedDate;
+            expected.sanitizeForSaving();
```

Comment:
I don't think this is the right time to introduce phantom type though. For one, the change is huge to apply across teammates, and to only do it here may introduce a lava flow issue. Secondly, this PR itself is continuing a half-solved issue and creating a new one here does not seem like a good idea.